modules/fb_instant_articles_display/fb_instant_articles_display.module

```diff
@@ -279,90 +279,152 @@ function fb_instant_articles_display_get_node_layout_settings($bundle_name) {
+  $formatters = fb_instant_articles_get_field_formatter_types();
   $formats = array();
-  // Header only elements
-  $formats['fbia_subtitle_formatter'] = array(
-    'label' => t('FBIA Subtitle'),
-    'field types' => array('text', 'text_long', 'text_with_summary'),
-  );
-  $formats['fbia_kicker_formatter'] = array(
-    'label' => t('FBIA Kicker'),
-    'field types' => array('text', 'text_long', 'text_with_summary'),
-  );
-  $formats['fbia_blockquote_formatter'] = array(
-    'label' => t('FBIA Blockquote'),
-    'field types' => array('text', 'text_long', 'text_with_summary'),
-  );
-  $formats['fbia_pullquote_formatter'] = array(
-    'label' => t('FBIA Pullquote'),
-    'field types' => array('text', 'text_long', 'text_with_summary'),
-  );
-  $formats['fbia_social_formatter'] = array(
-    'label' => t('FBIA Social Embed'),
-    'field types' => array('text', 'text_long', 'text_with_summary'),
-  );
-  $formats['fbia_credits_formatter'] = array(
-    'label' => t('FBIA Credits'),
-    'field types' => array('text', 'text_long', 'text_with_summary'),
-  );
-  $formats['fbia_copyright_formatter'] = array(
-    'label' => t('FBIA Copyright'),
-    'field types' => array('text', 'text_long', 'text_with_summary'),
-  );
-  $formats['fbia_author_formatter'] = array(
-    'label' => t('FBIA Author'),
-    'field types' => array('text', 'text_with_summary', 'list_text'),
-  );
-  $formats['fbia_video_formatter'] = array(
-    'label' => t('FBIA Video'),
-    'field types' => array('file'),
-  );
-  $formats['fbia_ad_formatter'] = array(
-    'label' => t('FBIA Ad'),
-    'field types' => array('text', 'text_long', 'text_with_summary'),
-    'settings'  => array(
-      'height' => '50',
-      'width' => '320',
-      'source' => 'url',
-    ),
-  );
-  $formats['fbia_analytics_formatter'] = array(
-    'label' => t('FBIA Analytics'),
-    'field types' => array('text', 'text_long', 'text_with_summary'),
-    'settings'  => array(
-      'source' => 'url',
-    ),
-  );
-  $formats['fbia_image_formatter'] = array(
-    'label' => t('FBIA Image'),
-    'field types' => array('image'),
-    'settings'  => array(
-      'style' => 'medium',
-      'caption' => '',
-      'likes' => '',
-      'comments' => '',
-      'fullscreen' => '',
-    ),
-  );
-  $formats['fbia_interactive_formatter'] = array(
-    'label' => t('FBIA Interactive'),
-    'field types' => array('text', 'text_with_summary'),
-    'settings'  => array(
-      'height' => '50',
-      'width' => 'no-margin',
-    ),
-  );
-  $formats['fbia_list_formatter'] = array(
-    'label' => t('FBIA List'),
-    'field types' => array('list_text', 'list_integer', 'list_float'),
-    'settings'  => array(
-      'list_type' => 'ol',
-    ),
-  );
-  $formats['fbia_transformer_formatter'] = array(
-    'label' => t('FBIA Transfomer'),
-    'field types' => array('text', 'text_long', 'text_with_summary'),
-  );
+  foreach ($formatters as $format) {
```

Comment:
A general note that I am not exactly crazy about this change I made here. I think I was headed down a road mentally that I thought would be saving manually entering in displays, but it didn't turn out that way. This whole switch could be left out in favor of the orignal array declaration.